File: elixir/two-fer/two_fer_test.exs
if !System.get_env("EXERCISM_TEST_EXAMPLES") do
  Code.load_file("two_fer.exs", __DIR__)
end

ExUnit.start()
ExUnit.configure(exclude: :pending, trace: true)

defmodule TwoFerTest do
  use ExUnit.Case

  test "no name given" do
    assert TwoFer.two_fer() == "One for you, one for me"
  end

  test "a name given" do
    assert TwoFer.two_fer("Gilberto Barros") == "One for Gilberto Barros, one for me"
  end

  test "when the parameter is a number" do
    assert_raise FunctionClauseError, fn ->
      TwoFer.two_fer(10)
    end
  end

  test "when the parameter is an atom" do
    assert_raise FunctionClauseError, fn ->
      TwoFer.two_fer(:bob)
    end
  end

  test "when the parameter is a charlist" do
    assert_raise FunctionClauseError, fn ->
      refute TwoFer.two_fer('Jon Snow')
    end
  end
end

File: scala/two-fer/src/test/scala/TwoferTest.scala
import org.scalatest.{Matchers, FunSuite}

/** @version 1.2.0 */
class TwoferTest extends FunSuite with Matchers {

  test("no name given") {
    Twofer.twofer() should be ("One for you, one for me.")
  }

  test("a name given") {
    Twofer.twofer("Alice") should be ("One for Alice, one for me.")
  }

  test("another name given") {
    Twofer.twofer("Bob") should be ("One for Bob, one for me.")
  }
}